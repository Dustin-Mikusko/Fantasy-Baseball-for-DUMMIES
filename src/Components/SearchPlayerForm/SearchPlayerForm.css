.search-player-form {
  height: 20px;
  border: none;
}

.player-search-input {
  font-size: 1.2em;
  margin-right: 10px;
  padding-left: 5px;
  margin-top: 150px;
  margin-bottom: 150px;
}

.player-search-btn {
  background-color: #041F42;
  color: white;
  font-size: 1.2em;
  border: none;
  border-radius: 3px;
}

.player-search-btn:hover {
  background-color: rgb(6, 71, 155);
  cursor: pointer;
}

.player-search-btn:active {
  transform: scale(.97)
}

.player-search-error {
  color: red;
  /* margin-top: 200px; */
}

.single-player {
  margin-top: 240px;
}
