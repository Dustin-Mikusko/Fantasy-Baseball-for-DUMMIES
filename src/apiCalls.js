export const fetchSingleName = name => {
  const options = {
    "method": "GET",
    "headers": {
      "x-rapidapi-host": "mlb-data.p.rapidapi.com",
      "x-rapidapi-key": "1b5b2533a1msh5cb04fc7ae14a5fp1cb4bdjsnca3f21f97e9d"
    }
  };
  const urlName = `${name}%25`;
  return fetch(`http://lookup-service-prod.mlb.com/json/named.search_player_all.bam?sport_code='mlb'&active_sw='Y'&name_part='${urlName}'`, options)
    .then(res => {
      if (!res.ok) {
        console.log('oh hi')
        throw Error('Uh oh! Player not found or player is no longer active')
      }
      console.log('oh hello')
      return res.json();
    })
}

export const fetchFullName = name => {
  const options = {
    "method": "GET",
    "headers": {
      "x-rapidapi-host": "mlb-data.p.rapidapi.com",
      "x-rapidapi-key": "1b5b2533a1msh5cb04fc7ae14a5fp1cb4bdjsnca3f21f97e9d"
    }
  };
  return fetch(`http://lookup-service-prod.mlb.com/json/named.search_player_all.bam?sport_code='mlb'&active_sw='Y'&name_part='${name}'`, options)
    .then(res => {
      if (!res.ok) {
        throw Error('Uh Oh! Player not found or player is no longer active')
      }
      return res.json();
    })
}

export const fetchSeasonHittingStats = (year, id) => {
  const options = {
    "method": "GET",
    "headers": {
      "x-rapidapi-host": "mlb-data.p.rapidapi.com",
      "x-rapidapi-key": "1b5b2533a1msh5cb04fc7ae14a5fp1cb4bdjsnca3f21f97e9d"
    }
  };
  return fetch(`http://lookup-service-prod.mlb.com/json/named.sport_hitting_tm.bam?league_list_id='mlb'&game_type='R'&season=${year}&player_id=${id}`, options)
    .then(res => {
      if (!res.ok) {
        throw Error('Uh Oh! Player not found or player is no longer active')
      }
      return res.json();
    })
}

export const fetchTop25 = (year, stat) => {
  const options = {
    "method": "GET",
    "headers": {
      "x-rapidapi-host": "mlb-data.p.rapidapi.com",
      "x-rapidapi-key": "1b5b2533a1msh5cb04fc7ae14a5fp1cb4bdjsnca3f21f97e9d"
    }
  };
  return fetch(`http://lookup-service-prod.mlb.com/json/named.leader_hitting_repeater.bam?sport_code='mlb'&results=25&game_type='R'&season=${year}&sort_column=${stat}&leader_hitting_repeater.col_in=rbi&leader_hitting_repeater.col_in=player_id&leader_hitting_repeater.col_in=name_display_first_last&leader_hitting_repeater.col_in=team_name`, options)
    .then(res => {
        if (!res.ok) {
          throw Error('Opps! There was an error on the play..')
        }
        return res.json();
    })
}


