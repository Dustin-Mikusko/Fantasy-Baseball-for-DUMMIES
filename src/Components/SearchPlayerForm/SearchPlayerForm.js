import React, { Component } from 'react';
import './SearchPlayerForm.css';
import { fetchSingleName, fetchFullName } from '../../apiCalls';
import PlayerPage from '../PlayerPage/PlayerPage';
import Header from '../Header/Header';
import PlayerContainer from '../PlayerContainer/PlayerContainer';

export class SearchPlayerForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      inputError: '',
      errorMessage: '',
      searchedPlayer: null,
      searchedPlayerList: null,
    }
  }

  handleChange = e => {
    this.setState({ name: e.target.value})
  }

  cleanSinglePlayer = data => {
    return {
      id: data.search_player_all.queryResults.row.player_id,
      name: data.search_player_all.queryResults.row.name_display_first_last,
      team: data.search_player_all.queryResults.row.team_full,
      position: data.search_player_all.queryResults.row.position,
      inches: data.search_player_all.queryResults.row.height_inches,
      feet: data.search_player_all.queryResults.row.height_feet,
      weight: data.search_player_all.queryResults.row.weight,
      bats: data.search_player_all.queryResults.row.bats,
      throws: data.search_player_all.queryResults.row.throws
    };
  };

  cleanListOfPlayers = players => {
    return players.map(player => {
      return {
        id: player.player_id,
        name: player.name_display_first_last,
        team: player.team_full,
        position: player.position,
        inches: player.height_inches,
        feet: player.height_feet,
        weight: player.weight,
        bats: player.bats,
        throws: player.throws
      };
    })
  }

  findPlayers = () => {
    const name = this.state.name;
    if (name.includes(' ')) {
      fetchFullName(name)
        .then(data => {
          if (data.search_player_all.queryResults.totalSize === '0') {
            this.setState({ errorMessage: "Oops! Player not found or is no longer active.."})
          } else {
            this.setState({ searchedPlayerList: [] })
            this.setState({ errorMessage: '' });
            const player = this.cleanSinglePlayer(data);
            this.setState({ searchedPlayer: player })
          }
        })
        .catch(err => this.setState({ errorMessage: err }))
    } else {
      fetchSingleName(name)
        .then(data => {
          if (data.search_player_all.queryResults.totalSize === '0') {
            this.setState({ errorMessage: "Oops! Player not found or is no longer active.."})
          } else {
            this.setState({ errorMessage: '' });
            this.setState({ searchedPlayer: null })
            const players = this.cleanListOfPlayers(data.search_player_all.queryResults.row)
            this.setState({ searchedPlayerList: players })
          } 
        })
        .catch(err => this.setState({ errorMessage: err }))
    }
  }

  render() {
    return (
      <>
        <Header />
        <form className='search-player-form'>
          <input type='text' className='player-search-input' onChange={this.handleChange} placeholder='e.g jose altuve'></input>
          <button type="button" className='player-search-btn'onClick={this.findPlayers}>SEARCH</button>
        </form>
        {this.state.errorMessage && <p>{this.state.errorMessage}</p>}
        {this.state.searchedPlayer && <div className='single-player'><PlayerPage 
          player={this.state.searchedPlayer}
        /></div>}
        {this.state.searchedPlayerList && <PlayerContainer 
        players={this.state.searchedPlayerList}
        />}
      </>
    )
  }
};

export default SearchPlayerForm;
