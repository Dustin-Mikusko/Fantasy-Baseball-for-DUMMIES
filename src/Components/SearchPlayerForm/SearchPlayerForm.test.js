import React from 'react';
import { SearchPlayerForm } from './SearchPlayerForm';
import { shallow } from 'enzyme';
import { fetchFullName, fetchSingleName } from '../../apiCalls';

jest.mock('../../apiCalls.js');

describe('SearchPlayerForm', () => {
  let wrapper;

  beforeEach(() => {
    fetchFullName.mockImplementation(() => {
      return Promise.resolve({
        "search_player_all": {
            "copyRight": " Copyright 2019 MLB Advanced Media, L.P.  Use of any content on this page acknowledges agreement to the terms posted here http://gdx.mlb.com/components/copyright.txt  ",
            "queryResults": {
                "created": "2020-01-12T21:44:24",
                "totalSize": "1",
                "row": {
                    "position": "3B",
                    "birth_country": "USA",
                    "weight": "180",
                    "birth_state": "NM",
                    "name_display_first_last": "Alex Bregman",
                    "college": "Louisiana State",
                    "height_inches": "0",
                    "name_display_roster": "Bregman",
                    "sport_code": "mlb",
                    "bats": "R",
                    "name_first": "Alex",
                    "team_code": "hou",
                    "birth_city": "Albuquerque",
                    "height_feet": "6",
                    "pro_debut_date": "2016-07-25T00:00:00",
                    "team_full": "Houston Astros",
                    "team_abbrev": "HOU",
                    "birth_date": "1994-03-30T00:00:00",
                    "throws": "R",
                    "league": "AL",
                    "name_display_last_first": "Bregman, Alex",
                    "position_id": "5",
                    "high_school": "Albuquerque Academy, Albuquerque, NM",
                    "name_use": "Alex",
                    "player_id": "608324",
                    "name_last": "Bregman",
                    "team_id": "117",
                    "service_years": "",
                    "active_sw": "Y"
                  }
              }
          }
      })
    });
    fetchSingleName.mockImplementation(() => {
      return Promise.resolve([
        {
          "position": "3B",
          "birth_country": "USA",
          "weight": "180",
          "birth_state": "NM",
          "name_display_first_last": "Alex Bregman",
          "college": "Louisiana State",
          "height_inches": "0",
          "name_display_roster": "Bregman",
          "sport_code": "mlb",
          "bats": "R",
          "name_first": "Alex",
          "team_code": "hou",
          "birth_city": "Albuquerque",
          "height_feet": "6",
          "pro_debut_date": "2016-07-25T00:00:00",
          "team_full": "Houston Astros",
          "team_abbrev": "HOU",
          "birth_date": "1994-03-30T00:00:00",
          "throws": "R",
          "league": "AL",
          "name_display_last_first": "Bregman, Alex",
          "position_id": "5",
          "high_school": "Albuquerque Academy, Albuquerque, NM",
          "name_use": "Alex",
          "player_id": "608324",
          "name_last": "Bregman",
          "team_id": "117",
          "service_years": "",
          "active_sw": "Y"
        }
      ])
    })
    wrapper = shallow(<SearchPlayerForm />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should change name state when handleChange is called', () => {
      const mockEvent = {
        target: {
          value: 'jose altuve'
        }
      };

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('name')).toEqual('jose altuve');
    });
  });

  describe('cleanSinglePlayer', () => {
    it('should return a player object', () => {
      const mockData = {
        "search_player_all": {
            "copyRight": " Copyright 2019 MLB Advanced Media, L.P.  Use of any content on this page acknowledges agreement to the terms posted here http://gdx.mlb.com/components/copyright.txt  ",
            "queryResults": {
                "created": "2020-01-12T21:44:24",
                "totalSize": "1",
                "row": {
                    "position": "3B",
                    "birth_country": "USA",
                    "weight": "180",
                    "birth_state": "NM",
                    "name_display_first_last": "Alex Bregman",
                    "college": "Louisiana State",
                    "height_inches": "0",
                    "name_display_roster": "Bregman",
                    "sport_code": "mlb",
                    "bats": "R",
                    "name_first": "Alex",
                    "team_code": "hou",
                    "birth_city": "Albuquerque",
                    "height_feet": "6",
                    "pro_debut_date": "2016-07-25T00:00:00",
                    "team_full": "Houston Astros",
                    "team_abbrev": "HOU",
                    "birth_date": "1994-03-30T00:00:00",
                    "throws": "R",
                    "league": "AL",
                    "name_display_last_first": "Bregman, Alex",
                    "position_id": "5",
                    "high_school": "Albuquerque Academy, Albuquerque, NM",
                    "name_use": "Alex",
                    "player_id": "608324",
                    "name_last": "Bregman",
                    "team_id": "117",
                    "service_years": "",
                    "active_sw": "Y"
                  }
              }
          }
      };
      const expected = {
        id: '608324',
        name: 'Alex Bregman',
        team: 'Houston Astros',
        position: '3B',
        inches: '0',
        feet: '6',
        weight: '180',
        bats: 'R',
        throws: 'R'
      };
      expect(wrapper.instance().cleanSinglePlayer(mockData)).toEqual(expected);
    });
  });

  describe('cleanListOfPlayers', () => {
    it('should take an array of players and return an array of player objects', () => {
      const mockData = [{
        "position": "3B",
        "birth_country": "USA",
        "weight": "180",
        "birth_state": "NM",
        "name_display_first_last": "Alex Bregman",
        "college": "Louisiana State",
        "height_inches": "0",
        "name_display_roster": "Bregman",
        "sport_code": "mlb",
        "bats": "R",
        "name_first": "Alex",
        "team_code": "hou",
        "birth_city": "Albuquerque",
        "height_feet": "6",
        "pro_debut_date": "2016-07-25T00:00:00",
        "team_full": "Houston Astros",
        "team_abbrev": "HOU",
        "birth_date": "1994-03-30T00:00:00",
        "throws": "R",
        "league": "AL",
        "name_display_last_first": "Bregman, Alex",
        "position_id": "5",
        "high_school": "Albuquerque Academy, Albuquerque, NM",
        "name_use": "Alex",
        "player_id": "608324",
        "name_last": "Bregman",
        "team_id": "117",
        "service_years": "",
        "active_sw": "Y"
    }];
      const expected = [{
        id: '608324',
        name: 'Alex Bregman',
        team: 'Houston Astros',
        position: '3B',
        inches: '0',
        feet: '6',
        weight: '180',
        bats: 'R',
        throws: 'R'
      }];
      expect(wrapper.instance().cleanListOfPlayers(mockData)).toEqual(expected);
    });
  });

  describe('findPlayers', () => {
    let mockState;
    beforeEach(() => {
      mockState = {
        name: 'jose altuve'
      };
    });

    it('should invoke fetchFullName if the name state has two names', () => {
      wrapper.setState(mockState);
      wrapper.instance().findPlayers();
      expect(fetchFullName).toHaveBeenCalled();
    });

    it('should set searchedPlayerList to empty array', async () => {
      wrapper.setState(mockState);
      await wrapper.instance().findPlayers();

      expect(wrapper.state('searchedPlayerList')).toEqual([])
    });

    it('should set ErrorMessage state to empty string', async () => {
      wrapper.setState(mockState);
      await wrapper.instance().findPlayers();
      wrapper.instance().forceUpdate();

      expect(wrapper.state('errorMessage')).toEqual('')
    });

    it('should set the searchPlayer to a player object', async () => {
      const mockPlayer = {
        id: '608324',
        name: 'Alex Bregman',
        team: 'Houston Astros',
        position: '3B',
        inches: '0',
        feet: '6',
        weight: '180',
        bats: 'R',
        throws: 'R'
      }
      wrapper.setState(mockState);
      await wrapper.instance().findPlayers();

      expect(wrapper.state('searchedPlayer')).toEqual(mockPlayer)
    })

    it('should set the errorMessage state if searched player is not found', async () => {
      fetchFullName.mockImplementation(() => {
        return Promise.resolve({
          "search_player_all": {
              "copyRight": " Copyright 2019 MLB Advanced Media, L.P.  Use of any content on this page acknowledges agreement to the terms posted here http://gdx.mlb.com/components/copyright.txt  ",
              "queryResults": {
                  "created": "2020-01-13T21:40:03",
                  "totalSize": "0"
              }
          }
      })
      });
      wrapper.setState(mockState);
      await wrapper.instance().findPlayers();
      expect(wrapper.state('errorMessage')).toEqual("Oops! Player not found or is no longer active..")
    });

    it('should invoke fetchSingleName if only one name is typed into the form', () => {
      const mockState = {
        name: 'jose'
      }
      wrapper.setState(mockState);
      wrapper.instance().findPlayers();
      expect(fetchSingleName).toHaveBeenCalled();
    });

    it('should set errorMessage state to empty string', async () => {
      const mockState = {
        name: 'jose'
      };

      wrapper.setState(mockState);
      await wrapper.instance().findPlayers();

      expect(wrapper.state('errorMessage')).toEqual('');
    });

    it('should set searchedPlayer state to null', async () => {
      const mockState = {
        name: 'jose'
      };

      wrapper.setState(mockState);
      await wrapper.instance().findPlayers();

      expect(wrapper.state('searchedPlayer')).toEqual(null);
    });

    it('should set the searchedPlayerList to an array of players',async () => {
      const mockState = {
        name: 'jose'
      };
      const expected = [
        {
          id: '608324',
          name: 'Alex Bregman',
          team: 'Houston Astros',
          position: '3B',
          inches: '0',
          feet: '6',
          weight: '180',
          bats: 'R',
          throws: 'R'
        }
      ]
      wrapper.setState(mockState);
      await wrapper.instance().findPlayers();

      expect(wrapper.state('searchedPlayerList')).toEqual(null);
    })
  });
})
