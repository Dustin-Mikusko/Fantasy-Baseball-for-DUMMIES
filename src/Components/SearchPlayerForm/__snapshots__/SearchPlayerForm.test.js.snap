// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPlayerForm should match the snapshot 1`] = `ShallowWrapper {}`;
